src/jolt/instruction/add.rs

```diff
@@ -0,0 +1,55 @@
+// use core::slice::SlicePattern;
+use ark_ff::PrimeField;
+use ark_std::log2;
+
+use super::JoltInstruction;
+use crate::jolt::subtable::{iden::IDENSubtable, lowerk::LOWERKSubtable, LassoSubtable};
+use crate::utils::instruction_utils::{chunk_and_concatenate_operands, concatenate_lookups, add_and_chunk_operands};
+
+#[derive(Copy, Clone, Default, Debug)]
+pub struct ADDInstruction(pub u64, pub u64);
+
+impl JoltInstruction for ADDInstruction {
+  fn combine_lookups<F: PrimeField>(&self, vals: &[F], C: usize, M: usize) -> F {
+    // The first C are from IDEN and the last C are from LOWER9
+    assert!(vals.len() == 2 * C);
+
+    // The output is the LOWER9(most significant chunk) || IDEN of other chunks
+    concatenate_lookups([&vals[C..(C+1)], &vals[1..C]].concat().as_slice(), C, M)
+  }
+
+  fn g_poly_degree(&self, _: usize) -> usize {
+    1
+  }
+
+  fn subtables<F: PrimeField>(&self) -> Vec<Box<dyn LassoSubtable<F>>> {
+    vec![Box::new(IDENSubtable::new()), Box::new(LOWERKSubtable::new(9))]
+  }
```

Comment:
sorry I wasn't clear –– I meant that this "9" should be calculated from the word size, `C`, and `M` (which requires passing `C` and `M` as parameters to this function)

```suggestion
  fn subtables<F: PrimeField>(&self, C: usize, M: usize) -> Vec<Box<dyn LassoSubtable<F>>> {
    let k: usize = 64 % (log2(M) / 2);
    vec![Box::new(IDENSubtable::new()), Box::new(LOWERKSubtable::new(k))]
  }
```